rosetta-source/src/main/rosetta/product-common-settlement-type.rosetta

```diff
@@ -300,11 +300,13 @@ type PriceQuantity: <"Defines a settlement as an exchange between two parties of
 	settlementTerms SettlementTerms (0..1) <"Whether the settlement is cash or physical and the corresponding terms. Attribute is optional in case the price/quantity settlement is defined as part of the product mechanics.">
 	effectiveDate AdjustableOrRelativeDate (0..1) <"Specifies the date at which the price and quantity become effective. This day may be subject to adjustment in accordance with a business day convention, or could be specified as relative to a trade date, for instance. Optional cardinality, as the effective date is usually specified in the product definition, so it may only need to be specified as part of the PriceQuantity in an increase/decrease scenario for an existing trade.">
 
-	condition RateOptionObservable: <"When the observable is a rate option, the price type must correspond to one of the interest rate price expressions.">
+	condition RateOptionObservable: <"When the observable is a rate option, the price type must be interest rate and the arithmetic operator must be specified.">
 	    if observable -> rateOption exists and price exists
-	    then 
-			price -> priceType contains PriceTypeEnum -> InterestRate
-			or price -> arithmeticOperator contains ArithmeticOperationEnum -> Multiply
+	    then price -> priceType all = PriceTypeEnum -> InterestRate and price -> arithmeticOperator exists
```

Comment:
Should this be:
`price extract [ priceType = PriceTypeEnum -> InterestRate and arithmeticOperator exists ] all = True`

Reply:
> price extract [ priceType = PriceTypeEnum -> InterestRate and arithmeticOperator exists ] all = True

This breaks python code generation, I can move to a separate function, and call the function from the condition

Reply:
> I can move to a separate function, and call the function from the condition
If you do this, will you be able to express the function's logic in Rosetta to work around the code generation issue, or will the function have to be implemented in java?

Reply:
The function logic can still be expressed in Rosetta.  Moving it to a function works because the python code gen doesn't attempt to generate for functions.